plugins/source/gcp/client/spec/schema.json

```diff
@@ -1,159 +1,127 @@
 {
   "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "$id": "https://github.com/cloudquery/cloudquery/plugins/source/gcp/client/spec/spec",
   "$ref": "#/$defs/Spec",
   "$defs": {
-    "Spec": {
-      "title": "CloudQuery GCP Source Plugin Spec",
+    "CredentialsConfig": {
+      "properties": {
+        "target_principal": {
+          "type": "string",
+          "format": "email"
+        },
+        "scopes": {
+          "items": {
+            "type": "string",
+            "pattern": "^https://www.googleapis.com/auth/(.)+$"
+          },
+          "type": "array",
+          "default": [
+            "https://www.googleapis.com/auth/cloud-platform"
+          ]
+        },
+        "delegates": {
+          "items": {
+            "type": "string",
+            "format": "email"
+          },
+          "type": "array"
+        },
+        "subject": {
+          "type": "string",
+          "minLength": 1
+        }
+      },
+      "additionalProperties": false,
       "type": "object",
+      "required": [
+        "target_principal"
+      ]
+    },
+    "Spec": {
+      "not": {
+        "required": [
+          "project_filter",
+          "folder_ids"
+        ]
+      },
       "properties": {
         "project_ids": {
-          "title": "Project IDs",
-          "description": "GCP projects to connect to. By default will use all projects available to the current authenticated account",
-          "type": "array",
           "items": {
             "type": "string",
             "minLength": 1
-          }
+          },
+          "type": "array"
         },
         "folder_ids": {
-          "title": "Folder IDs",
-          "description": "Location of the projects to sync from. Mutually exclusive with project_filter",
```

Comment:
What happened to all the descriptions?

Reply:
I removed them (they can be added via `jsonschema_description` tag, but the code lines would look too long).

I guess we can experiment with the descriptions & generating docs from the schema later, but for now, the descriptions don't give any additional info to the validation errors, so I suppose it's OK to omit them.